packages/flutter/lib/src/widgets/image.dart

```diff
@@ -566,6 +602,52 @@ class Image extends StatefulWidget {
   /// (false), when the image provider changes.
   final bool gaplessPlayback;
 
+  /// A builder that specifies the widget to display to the user while the
+  /// image provider's stream is still loading its image.
+  ///
+  /// The widget returned from this builder will be sized to match the size of
+  /// the image.
+  ///
+  /// If this is `null`, and the image is loaded incrementally (e.g. over a
+  /// network), the user will receive no indication of the progress as the
+  /// bytes of the image are loaded.
+  ///
+  /// Once an image has fully loaded, this builder will no longer be consulted.
+  ///
+  /// {@tool sample}
```

Comment:
You could make this a snippet pretty easily...

Reply:
Done